cirq/linalg/decompositions.py

```diff
@@ -264,12 +264,115 @@ def so4_to_magic_su2s(
     return a, b
 
 
-def kak_canonicalize_vector(
-        x: float, y: float, z: float
-) -> Tuple[complex,
-           Tuple[np.ndarray, np.ndarray],
-           Tuple[float, float, float],
-           Tuple[np.ndarray, np.ndarray]]:
+class KakDecomposition:
+    """A convenient description of an arbitrary two-qubit operation.
+
+    Any two qubit operation U = can be decomposed into the form
+
+        U = g · (a1 ⊗ a0) · exp(i·(x·XX + y·YY + z·ZZ)) · (b1 ⊗ b0)
+
+    This class stores g, (b0, b1), (x, y, z), and (a0, a1).
```

Comment:
Would it be worth linking to a page (or the paper you linked to https://arxiv.org/abs/quant-ph/0507171) for more information?

Reply:
Done.